src/utils/upload/index.ts

```diff
@@ -120,3 +131,60 @@ export function areFileWithCollectionsSame(
 ): boolean {
     return firstFile.localID === secondFile.localID;
 }
+
+export function analyseUploadFiles(
+    uploadType: UPLOAD_TYPE,
+    toUploadFiles: File[] | ElectronFile[]
+): AnalysisResult {
+    if (isElectron() && uploadType === UPLOAD_TYPE.FILES) {
+        return NULL_ANALYSIS_RESULT;
+    }
+
+    const paths: string[] = toUploadFiles.map((file) => file['path']);
+    const getCharCount = (str: string) => (str.match(/\//g) ?? []).length;
+    paths.sort((path1, path2) => getCharCount(path1) - getCharCount(path2));
+    const firstPath = paths[0];
+    const lastPath = paths[paths.length - 1];
+
+    const L = firstPath.length;
+    let i = 0;
+    const firstFileFolder = firstPath.substring(0, firstPath.lastIndexOf('/'));
+    const lastFileFolder = lastPath.substring(0, lastPath.lastIndexOf('/'));
+    while (i < L && firstPath.charAt(i) === lastPath.charAt(i)) i++;
+    let commonPathPrefix = firstPath.substring(0, i);
+
+    if (commonPathPrefix) {
+        commonPathPrefix = commonPathPrefix.substring(
+            0,
+            commonPathPrefix.lastIndexOf('/')
+        );
+        if (commonPathPrefix) {
+            commonPathPrefix = commonPathPrefix.substring(
+                commonPathPrefix.lastIndexOf('/') + 1
+            );
+        }
+    }
+    return {
+        suggestedCollectionName: commonPathPrefix || null,
+        multipleFolders: firstFileFolder !== lastFileFolder,
+    };
+}
+export function getCollectionWiseFiles(toUploadFiles: File[] | ElectronFile[]) {
+    const collectionWiseFiles = new Map<string, (File | ElectronFile)[]>();
+    for (const file of toUploadFiles) {
+        const filePath = file['path'] as string;
+
+        let folderPath = filePath.substring(0, filePath.lastIndexOf('/'));
+        if (folderPath.endsWith(METADATA_FOLDER_NAME)) {
```

Comment:
Didn't follow what are we doing here. Should we add a comment explaining what's happening here and may be why too if necessary?

Reply:
rename METADATA_FOLDER_NAME to ENTE_METADAT_FOLDER

Reply:
verify if all files are json

Reply:
didn't do the verify JSON files 😅  ...was a big change